builtin/providers/aws/resource_aws_cloudwatch_logs_subscription_filter.go

```diff
@@ -0,0 +1,341 @@
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"role": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+			},
+		},
+	}
+}
+
+func resourceAwsCloudwatchLogsSubscriptionFilterCreate(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).cloudwatchlogsconn
+
+	name := d.Get("name").(string)
+
+	log_group := d.Get("log_group").(string)
+
+	destination := d.Get("destination").(string)
+	if strings.HasPrefix(destination, "arn:aws:kinesis:") {
+		destination_arn_sliced := strings.Split(destination, "/")
+		destination_name := destination_arn_sliced[len(destination_arn_sliced)-1]
+
+		kinesis_conn := meta.(*AWSClient).kinesisconn
+		waitForKinesisStreamToActivate(kinesis_conn, destination_name)
+	} else if strings.HasPrefix(destination, "arn:aws:lambda") {
+		lambda_conn := meta.(*AWSClient).lambdaconn
+
+		lambda_arn_sliced := strings.Split(destination, ":")
+		function_name := lambda_arn_sliced[len(lambda_arn_sliced)-1]
+		statement_id := lambdaPermissionStatementId(log_group, destination)
+
+		if !permissionExists(function_name, statement_id, lambda_conn) {
+			region := lambda_arn_sliced[3]
+			accountid := lambda_arn_sliced[4]
+			principal := fmt.Sprintf("logs.%s.amazonaws.com", region)
+			source_arn := fmt.Sprintf("arn:aws:logs:%s:%s:log-group:%s:*", region, accountid, log_group)
+
+			params := &lambda.AddPermissionInput{
+				Action:        aws.String("lambda:InvokeFunction"),
+				FunctionName:  aws.String(function_name),
+				Principal:     aws.String(principal),
+				StatementId:   aws.String(statement_id),
+				SourceArn:     aws.String(source_arn),
+				SourceAccount: aws.String(accountid),
+			}
+
+			log.Printf("[DEBUG] Attempting: to do add-access with params \"%#v\"", params)
+			_, err := lambda_conn.AddPermission(params)
+			if err != nil {
+				if awsErr, ok := err.(awserr.Error); ok {
+					return fmt.Errorf("[WARN] Error doing add-access for LogGroup (%s) to lambda (%s), message: \"%s\", code: \"%s\"",
+						log_group, destination, awsErr.Message(), awsErr.Code())
+				} else {
+					return fmt.Errorf("Error creating Cloudwatch logs subscription filter %s: %#v", name, err)
+				}
+			}
+		}
+	}
+
+	params := getAwsCloudWatchLogsSubscriptionFilterInput(d)
+
+	log.Printf("[DEBUG] Creating SubscriptionFilter %#v", params)
+	sleep_randomizer := rand.New(rand.NewSource(time.Now().UnixNano()))
+
+	attemptCount := 1
+	for attemptCount <= CLOUDWATCH_LOGS_SUBSCRIPTION_FILTER_MAX_THROTTLE_RETRIES {
+		_, err := conn.PutSubscriptionFilter(&params)
+		attemptCount += 1
+		if err != nil {
+			if awsErr, ok := err.(awserr.Error); ok {
+				if awsErr.Code() == "InvalidParameterException" {
+					log.Printf("[DEBUG] Caught message: \"%s\", code: \"%s\". Attempt %d/%d: Sleeping for a bit to throttle back put request", awsErr.Message(), awsErr.Code(), attemptCount, CLOUDWATCH_LOGS_SUBSCRIPTION_FILTER_MAX_THROTTLE_RETRIES)
```

Comment:
instead of escaping all the quotes, you may use `%q` modifier which will enclose the string into quotes for you.